File: CourseProjects/CourseProjects/VisualStudioProjects/DesignPatternsExamples/TemplateMethod/HttpConnection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TemplateMethod {
    class HttpConnection : Connection {
        protected override void OpenConnection() {
            Console.WriteLine("HttpConnection.Open");
        }

        protected override void SendData(char[] data) {
            Console.WriteLine("HttpConnection.Send");
        }

        protected override void CloseConnection() {
            Console.WriteLine("HttpConnection.Close");
        }
    }
}

File: CourseProjects/CourseProjects/JavaEclipseProjects/DesignPatternsExamples/src/patterns/templatemethod/HttpConnection.java
package patterns.templatemethod;

public class HttpConnection extends Connection {
    void openConnection() {
        System.out.println("HttpConnection.openConnection");
    }
    void sendData(byte[] data) {
        System.out.println("HttpConnection.sendData");
    }
    void closeConnection() {
        System.out.println("HttpConnection.closeConnection");
    }
}
